pkgs/by-name/yo/yourkit-java/package.nix

```diff
@@ -0,0 +1,84 @@
+{ fetchzip
+, lib
+, stdenv
+, copyDesktopItems
+, imagemagick
+, makeDesktopItem
+, jre
+}:
+let
+  vPath = v: lib.elemAt (lib.splitString "-" v) 0;
+
+  arch = let inherit (stdenv.targetPlatform) system;
+         in if system == "x86_64-linux" then "x64"
+            else if system == "aarch64=linux" then "arm64"
+            else throw "Unsupported system";
+
+  version = "2023.9-b107";
+
+  desktopItem = makeDesktopItem {
+    name = "YourKit Java Profiler";
+    desktopName = "YourKit Java Profiler " + version;
+    type = "Application";
+    exec = "yourkit-java-profiler %f";
+    icon = "yourkit-java-profiler";
+    categories = [ "Development" "Java" "Profiling" ];
+    terminal = false;
+    startupWMClass = "YourKit Java Profiler";
+  };
+in
+stdenv.mkDerivation rec {
+  inherit version;
+
+  pname = "yourkit-java";
+
+  src = fetchzip {
+    url = "https://download.yourkit.com/yjp/${vPath version}/YourKit-JavaProfiler-${version}-${arch}.zip";
+    hash = "sha256-vlYmGAp8wURcNN+0Qxj+fWjLXwkAgkRnRdUMvg2TM+k=";
+  };
+
+  nativeBuildInputs = [ copyDesktopItems imagemagick ];
+
+  buildInputs = [ jre ];
+
+  desktopItems = [ desktopItem ];
+
+  installPhase = ''
+    runHook preInstall
+
+    mkdir -p $out
+    cp -pr bin lib license.html license-redist.txt probes samples $out
+    cp ${./forbid-desktop-item-creation} $out/bin/forbid-desktop-item-creation
+    for i in attach integrate; do
+        sed -i -e 's|profiler.sh|yourkit-java-profiler|' $out/bin/$i.sh
```

Comment:
Use `substituteInPlace`

Reply:
I'm sorry for the lateness. These are updated in the following changes.